### PR TITLE
Uncross accidentally crossed bonds.

### LINKED ISSUE #6170: Crossed bond not correctly drawn
**Describe the bug**
This looks like one of the edge cases in double bond drawing that @DavidACosgrove has been working on lately.
I ran into a quite peculiarly (let's put it this way) drawn molecule where double bonds are explicitly marked as a _Cis or trans (either) double bond_ and hence should be drawn as crossed bonds.
With the original coordinates this happens correctly only for one of them:

```
from rdkit import Chem
from rdkit.Chem.Draw import rdDepictor

mol = Chem.MolFromMolBlock("""
     RDKit          2D

  8  7  0  0  0  0  0  0  0  0999 V2000
    3.0428   -1.6819    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.0022    1.7243    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8934   -1.6549    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
    4.1949   -2.8653    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.9237    1.6929    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
    4.1521    2.9055    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6779   -1.6657    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.6267    1.6960    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  1  3  2  3
  1  4  1  0
  2  5  2  3
  2  6  1  0
  3  7  1  0
  5  8  1  0
M  END
""")
mol
```
image
After generating proper coordinates both double bonds are depicted correctly as crossed:
```
mol_copy = Chem.Mol(mol)
rdDepictor.Compute2DCoords(mol_copy)
0
mol_copy
```
image
It would be great if both double bonds could be drawn as crossed also if the starting coordinates are a bit peculiar.

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Fixes #6170

#### What does this implement/fix? Explain your changes.
If it detects that a double bond has been crossed at the initial calculation, it uncrosses it.

### CODE REVIEW

**Comment by DavidACosgrove** [comment]
The failure is in the Java diversity picker.  I'd be quite alarmed if my tweak in the graphics code has cause that!

**Comment by bp-kelley** [comment]
The java failure is in the diversity picker which seems unrelated to these changes.

**Comment by bp-kelley** [comment]
Trying a re-run of the java build.

**Comment by greglandrum** [comment]
> The java failure is in the diversity picker which seems unrelated to these changes. (agreeing with @DavidACosgrove here...)

Yes, these happen periodically due to random number collisions. It's intermittent enough that it's not worth fixing... just ignore them. :-)

File: Code/GraphMol/MolDraw2D/DrawMol.cpp [hunk 1]
```diff
@@ -2798,6 +2798,15 @@ void DrawMol::calcDoubleBondLines(double offset, const Bond &bond, Point2D &l1s,
       }
       bondNonRing(bond, offset, l2s, l2f);
     }
+
+    // Occasionally, as seen in Github6170, a bad geometry about a bond can
+    // result in the bonds being crossed as perpendiculars have become
+    // confused.  Usually this is the result of when a bond to the
+    // double bond is roughly linear with it.  This is a cheap test to see if
+    // this has happened, uncrossing them if necessary.
+    if (!areBondsLinear(l1s, l1f, l2f, l2s)) {
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawMol.cpp:2807]: I'm just realizing now what a poor name this is for this function: it really should be `areBondsParallel()`.
But that's not for this PR.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**